app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/view/RecentSyncedTab.kt

```diff
@@ -0,0 +1,227 @@
+        is RecentSyncedTabState.Success -> recentSyncedTabState.tab
+    }
+    RecentSyncedTabItem(
+        recentSyncedTab,
+        onRecentSyncedTabClick,
+        onSyncedTabSeeAllButtonClick,
+    )
+}
+
+/**
+ * A recent synced tab.
+ *
+ * @param tab Optional synced tab. If null, displays placeholders.
+ * @param onRecentSyncedTabClick Invoked when item is clicked.
+ * @param onSeeAllButtonClick Invoked when "See all" button is clicked.
+ */
+@Suppress("LongMethod")
+@Composable
+private fun RecentSyncedTabItem(
+    tab: RecentSyncedTab?,
+    onRecentSyncedTabClick: (RecentSyncedTab) -> Unit,
+    onSeeAllButtonClick: () -> Unit,
+) {
+    Card(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(180.dp)
+            .clickable { tab?.let { onRecentSyncedTabClick(tab) } },
+        shape = RoundedCornerShape(8.dp),
+        backgroundColor = FirefoxTheme.colors.layer2,
+        elevation = 6.dp
+    ) {
+        Column(modifier = Modifier.padding(16.dp)) {
+            Row(modifier = Modifier.height(IntrinsicSize.Min)) {
+                if (tab == null) {
+                    RecentTabImagePlaceholder()
+                } else {
+                    ThumbnailCard(
+                        url = tab.url,
+                        key = tab.url.hashCode().toString(),
+                        modifier = Modifier
+                            .size(108.dp, 80.dp)
+                            .clip(RoundedCornerShape(8.dp))
+                    )
+                }
+
+                Spacer(modifier = Modifier.width(16.dp))
+
+                Column(
+                    verticalArrangement = Arrangement.SpaceBetween,
+                    modifier = Modifier.fillMaxHeight()
+                ) {
+                    if (tab == null) {
+                        RecentTabTitlePlaceholder()
+                    } else {
+                        PrimaryText(
+                            text = tab.title,
+                            fontSize = 14.sp,
+                            maxLines = 2,
+                            overflow = TextOverflow.Ellipsis,
+                        )
+                    }
+
+                    Row(verticalAlignment = Alignment.CenterVertically) {
+                        if (tab == null) {
+                            Box(
+                                modifier = Modifier
+                                    .background(FirefoxTheme.colors.layer3)
+                                    .size(18.dp)
+                            )
+                        } else {
+                            Image(
+                                painter = painterResource(R.drawable.ic_synced_tabs),
+                                contentDescription = stringResource(
+                                    R.string.recent_tabs_synced_device_icon_content_description
+                                ),
+                                modifier = Modifier.size(18.dp, 18.dp)
```

Comment:
IIRC I think we can simplify this to `size(18.dp)` since the width and height are the same.